pkg/apis/batch/v1alpha1/job.go

```diff
@@ -61,16 +61,19 @@ type JobSpec struct {
 	MinAvailable int32 `json:"minAvailable,omitempty" protobuf:"bytes,2,opt,name=minAvailable"`
 
 	// The volumes mount on Job
+	// +patchStrategy=merge +patchMergeKey=mountPath
 	// +optional
-	Volumes []VolumeSpec `json:"volumes,omitempty" protobuf:"bytes,3,opt,name=volumes"`
+	Volumes []VolumeSpec `json:"volumes,omitempty" patchStrategy:"merge" patchMergeKey:"mountPath" protobuf:"bytes,3,opt,name=volumes"`
 
 	// Tasks specifies the task specification of Job
+	// +patchStrategy=merge +patchMergeKey=name
 	// +optional
-	Tasks []TaskSpec `json:"tasks,omitempty" protobuf:"bytes,4,opt,name=tasks"`
+	Tasks []TaskSpec `json:"tasks,omitempty" patchStrategy:"merge" patchMergeKey:"name" protobuf:"bytes,4,opt,name=tasks"`
 
 	// Specifies the default lifecycle of tasks
+	// +patchStrategy=merge +patchMergeKey=event
 	// +optional
-	Policies []LifecyclePolicy `json:"policies,omitempty" protobuf:"bytes,5,opt,name=policies"`
+	Policies []LifecyclePolicy `json:"policies,omitempty" patchStrategy:"merge" patchMergeKey:"event" protobuf:"bytes,5,opt,name=policies"`
```

Comment:
Can this take effect for CRD?

Reply:
sorry for late replay, it didn't when i tried, however, i decided to not use this feature and instead implement the my own logic. only because some of types are map and this package does not handle map replace with key.

Reply:
What's the original intention of modifying this field and how will you implement your logic?

Reply:
i reverted the changes for this field, so this field does not change any more. I have only introduced the patch field in this pr, implementation is done here: https://github.com/volcano-sh/volcano/pull/4491